src/store/getters.ts

```diff
@@ -0,0 +1,114 @@
+import ApplicationState from '@/types/ApplicationState'
+// @ts-ignore
+import { transformToRSQL } from '@molgenis/rsql'
+import Getters from '@/types/Getters'
+import Variant from '@/types/Variant'
+import Variable from '@/types/Variable'
+
+export default {
+  variants: (state: ApplicationState): Variant[] =>
+    state.variables.reduce((result: Variant[], variable: Variable): Variant[] =>
+      variable.variants.reduce((accumulator: Variant[], variant: Variant) =>
+        accumulator.some((candidate: Variant): boolean => candidate.id === variant.id)
+          ? accumulator
+          : [...accumulator, variant], result), []),
+  variantIds: (state: ApplicationState, getters: Getters): number[] =>
+    getters.variants.map(variant => variant.id),
+  rsql: (state: ApplicationState, getters: Getters) => {
+    let operands: Object[] = []
+    if (getters.variantIds.length > 0) {
+      operands.push({
+        selector: 'variant_id',
+        comparison: '=in=',
+        arguments: getters.variantIds
+      })
+    } else return ''
+    if (state.facetFilter.ageGroupAt1A.length > 0) {
```

Comment:
lost of code duplication maybe Object.keys and a foreach or map ?

Reply:
similar issue here, we have a discrepancy between the snake case attribute names and the camelcase property names